src/index.ts

```diff
@@ -54,11 +54,17 @@ app.doc('/open-api.json', {
 	}
 });
 
-app.route('/profiles', profileRoutes);
-app.route('/teams', teamRoutes);
+// Public routes
 app.route('/auth', authRoutes);
-
 // Handle static assets using Cloudflare Workers
 app.get('/assets/*', async (context: Context<EnvironmentBindings>) => context.env.ASSETS.fetch(context.req.raw));
 
+// Public routes
+app.route('/profiles', publicProfileRoutes);
+
+// Protected routes (after auth middleware)
+app.use('/*', authMiddleware);
+app.route('/profiles', protectedProfileRoutes);
+app.route('/teams', teamRoutes);
```

Comment:
```suggestion
app.route('/profiles', publicProfileRoutes);
app.route('/profiles', authMiddleware, protectedProfileRoutes);

app.route('/teams', authMiddleware, publicTeamRoutes);
app.route('/teams', authMiddleware, protectedTeamRoutes);
```

Being very nitpicky here: I'd rather we explicitly use middlewares than use it for all routes. It makes it easier to reason about the code.

Reply:
Yes, it would be better to reason. But, the `route()` requires max 2 parameters, so we can't use middleware inside `route()`
I don't know how to explicitly put middleware without `app.use('/*', authMiddleware);` 🤔 
Any ideas? @kbventures @bcha92

Reply:
Seems like Hono doesn't allow a third argument for `app.route()`.
However, it recommends using middleware above the handler:


If you have the middleware that you want to execute, write the code above the handler.
```jsx

app.use(logger())
app.get('/foo', (c) => c.text('foo'))
```

Hono doc [here](https://hono.dev/docs/api/routing#routing-priority).

Reply:
Seems like for routing in Hono, handlers and middleware are executed in registration order.
There is no third argument for `.route()`, but the `.use()` usage for middleware is a bit vague and is only mentioned sparingly in this [page](https://hono.dev/docs/api/routing#routing-priority).

Reply:
Well, if it would be impossible to explicitly use authMiddleware at `index.ts`, we should put authMiddleware in the `createRoute()` middleware property at `routes/profile/index.ts`

Reply:
Yeah, I thought Hono by itself would allow for an array of middlewares like express (that is actually a neat feature... oh well...)

But, the openapi thing we are using does [support an array of middlewares](https://github.com/honojs/middleware/tree/main/packages/zod-openapi#configure-middleware-for-each-endpoint), so maybe migrate to that soon-ish? 😅

Reply:
@lylaminju @madcampos  the max two parameters is why I went down that route.

Reply:
Conversation moved to https://github.com/torontojs/vms/issues/96